lib/CoreReplicator.js

```diff
@@ -0,0 +1,46 @@
+/**
+* @module CoreReplicator
+ */
+
+import { CoreIdCache } from "./authstore/coreIdCache"
+import { Sqlite } from "./sqlite"
+
+export class CoreReplicator {
+  /** @type {string} */
+  #coreId
+  #masterCore
+  /** @type {Corestore} */
+  #store
+  /** @type {CoreIdCache} */
+  #coreIdCache
+  /** @type {string} */
+  #extensionNamespace = 'replicator'
+  /** @type {any} */
+  #extension
+  /**
+* Create a CoreReplicator instance. This class is in charge of replicating cores belonging to a namespace in a corestore with other peers. This is acomplished by registering an hypercore extension.
+* @param {string} coreId - core public key to use as master core to register the extension in
+* @param {Corestore} store - set of cores to share
+* @param {CoreIdCache} coreIdCache  
+  */
+  constructor(coreId, store, coreIdCache){
+    this.#coreId = coreId
+    this.#store = store
+    this.#coreIdCache = coreIdCache
+    this.#masterCore = this.#store.get({key:this.#coreId})
+  }
+
+  /**
+* @param {StoreType} type
+  */
+  replicate(type){
```

Comment:
This name would get a bit confusing because of the existing `hypercore.replicate` method. We could instead use something like `CoreIdExtension` and maybe this method could be called `share`? I don't know if those ideas are very good, though. This is a difficult one to name, haha.